scripts/check_beaker

```diff
@@ -8,6 +8,33 @@ if [ ! -e "/var/lib/restraint/config.conf" ]; then
         . /etc/profile.d/beaker-harness-env.sh
     fi
 
+    # Try for up to 60 seconds to make sure system has an IP address
+    if command -v hostname; then
+        COUNT=0
+        # 12 tries * 5 seconds = Maximum 60 seconds
+        MAX_TRIES=12
+        # Loop until we have an IP address.
+        # 'hostname -I' will output something like: 192.168.1.1 if we have an
+        # IP address. If no IP address it will output a blank line.
+        IP_FAIL=""
+        while ! hostname -I | grep '[0-9]'
+        do
+            IP_FAIL=0
+            COUNT=$(( COUNT + 1 ))
+            echo "$0: Waiting for IP address. Attempt: ${COUNT} of ${MAX_TRIES} failed" >&2
+            if [[ ${COUNT} -ge ${MAX_TRIES} ]]; then
```

Comment:
Just a suggestion but you could simplify your code by using the bash magical "SECONDS" variable.  Before loop, initialize it to zero then "if [ $SECONDS -ge ${MAX_SECONDS} ]; then".

Reply:
Could do that. I guess I personally like the idea of N of X tries.